python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
         return SparseMatrix(numRows, numCols, colPtrs, rowIndices, values)
 
 
+class DistributedMatrix(object):
+    """Represents a distributively stored matrix backed by one or more RDDs."""
+    def numRows(self):
+        """Get or compute the number of rows."""
+        raise NotImplementedError
+
+    def numCols(self):
+        """Get or compute the number of cols."""
+        raise NotImplementedError
+
+
+class DistributedMatrices(object):
+    """Factory methods for distributed matrices."""
+    @staticmethod
+    def rowMatrix(rows, numRows=0, numCols=0):
+        """
+        Create a RowMatrix.
+
+        :param rows: An RDD of Vectors.
+        """
+        javaRowMatrix = callMLlibFunc("createRowMatrix", rows, long(numRows), int(numCols))
+        jrm = JavaModelWrapper(javaRowMatrix)
+        return RowMatrix(jrm)
+
+    @staticmethod
+    def indexedRowMatrix(rows, numRows=0, numCols=0):
+        """
+        Create an IndexedRowMatrix.
+
+        :param rows: An RDD of IndexedRows or (long, Vector) tuples.
+        """
+        # We use DataFrames for serialization of IndexedRows from Python, so convert the RDD to a
+        # DataFrame. This will convert each IndexedRow to a Row containing the 'index' and 'vector'
+        # values, which can both be easily serialized.  We will convert back to IndexedRows on the
+        # Scala side.
+        javaIndexedRowMatrix = callMLlibFunc("createIndexedRowMatrix", rows.toDF(),
```

Comment:
Thanks for the detailed explanation. But I was also thinking if it would be better to help in adding `IndexedRow` and `MatrixEntity` to SerDe so that it is more useful in the long run?